app/controllers/external_activities_controller.rb

```diff
@@ -288,6 +288,20 @@ def matedit
     end
   end
 
+  def archive
+    authorize @external_activity
+    @external_activity.archive!
+    flash[:notice]= t("matedit.archive_success", {name: @external_activity.name})
+    redirect_to :search  #TBD:  Where to go?
```

Comment:
The one problem with this at the moment, is that the flash message doesn't seem to display on the search page.   I am looking into it.